lib/Controller/UsersController.php

```diff
@@ -397,343 +371,60 @@ private function generateTokenAndSendMail($userId, $email) {
 	 * @return DataResponse
 	 */
 	public function create($username, $password, array $groups= [], $email='') {
-		if ($email !== '' && !$this->mailer->validateMailAddress($email)) {
-			return new DataResponse(
-				[
-					'message' => (string)$this->l10n->t('Invalid mail address')
-				],
-				Http::STATUS_UNPROCESSABLE_ENTITY
-			);
-		}
-
-		$currentUser = $this->userSession->getUser();
-
-		if (!$this->isAdmin) {
-			if (!empty($groups)) {
-				foreach ($groups as $key => $group) {
-					$groupObject = $this->groupManager->get($group);
-					if ($groupObject === null) {
-						unset($groups[$key]);
-						continue;
-					}
-
-					if (!$this->groupManager->getSubAdmin()->isSubAdminofGroup($currentUser, $groupObject)) {
-						unset($groups[$key]);
+		try {
+			$user = $this->createUserService->createUser(['username' => $username, 'password' => $password, 'email' => $email]);
+			if ($user instanceof User) {
```

Comment:
This check seems unneeded. The function will return an IUser and throw exceptions in case of error.
Note that, the app should be using the `IUser` interface, not the private implementation

Reply:
Removed the check. Also used IUser interface.